packages/frontend/src/components/SchedulerModals/SchedulerModalBase/SchedulerForm.tsx

```diff
@@ -1,17 +1,40 @@
-import { Button } from '@blueprintjs/core';
+import { Button, Colors, Icon } from '@blueprintjs/core';
+import { Tooltip2 } from '@blueprintjs/popover2';
 import cronstrue from 'cronstrue';
 import React, { FC, useMemo } from 'react';
+import useHealth from '../../../hooks/health/useHealth';
 import { useSlackChannels } from '../../../hooks/slack/useSlackChannels';
+import { useGetSlack } from '../../../hooks/useSlack';
 import { isInvalidCronExpression } from '../../../utils/fieldValidators';
 import { ArrayInput } from '../../ReactHookForm/ArrayInput';
 import AutoComplete from '../../ReactHookForm/AutoComplete';
 import Form from '../../ReactHookForm/Form';
 import Input from '../../ReactHookForm/Input';
-import { SlackIcon, TargetRow } from './SchedulerModalBase.styles';
+import { hasRequiredScopes } from '../../UserSettings/SlackSettingsPanel';
+import { EmailIcon, SlackIcon, TargetRow } from './SchedulerModalBase.styles';
+
+enum States {
+    LOADING,
+    SUCCESS,
+    ERROR,
+}
 
 const SchedulerForm: FC<
     { disabled: boolean } & React.ComponentProps<typeof Form>
 > = ({ disabled, methods, ...rest }) => {
+    const slackQuery = useGetSlack();
+    const state = useMemo(() => {
```

Comment:
If we only use the success state we can change this to `hasValidSlackClient`

Reply:
I have added both errors to the tooltip